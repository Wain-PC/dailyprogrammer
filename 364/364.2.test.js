const { solve } = require("./364.2");

describe("364.2", () => {
  const examples = [
    [1, 5, 7, 9, 9],
    [1, 2, 1, 2, 1, 0],
    [10, 12, 41, 62, 31, 50],
    [10, 12, 41, 62, 31]
  ];

  examples.forEach(example =>
    it(`should solve example ${example}`, () => {
      expect(solve(example)).toMatchSnapshot();
    })
  );
});
