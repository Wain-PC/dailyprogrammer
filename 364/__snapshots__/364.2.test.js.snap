// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`364.2 should solve example 1,2,1,2,1,0 1`] = `
Object {
  "counter": 3,
  "steps": Array [
    Array [
      1,
      2,
      1,
      2,
      1,
      0,
    ],
    Array [
      1,
      1,
      1,
      1,
      1,
      1,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
    ],
  ],
}
`;

exports[`364.2 should solve example 1,5,7,9,9 1`] = `
Object {
  "counter": 23,
  "steps": Array [
    Array [
      1,
      5,
      7,
      9,
      9,
    ],
    Array [
      4,
      2,
      2,
      0,
      8,
    ],
    Array [
      2,
      0,
      2,
      8,
      4,
    ],
    Array [
      2,
      2,
      6,
      4,
      2,
    ],
    Array [
      0,
      4,
      2,
      2,
      0,
    ],
    Array [
      4,
      2,
      0,
      2,
      0,
    ],
    Array [
      2,
      2,
      2,
      2,
      4,
    ],
    Array [
      0,
      0,
      0,
      2,
      2,
    ],
    Array [
      0,
      0,
      2,
      0,
      2,
    ],
    Array [
      0,
      2,
      2,
      2,
      2,
    ],
    Array [
      2,
      0,
      0,
      0,
      2,
    ],
    Array [
      2,
      0,
      0,
      2,
      0,
    ],
    Array [
      2,
      0,
      2,
      2,
      2,
    ],
    Array [
      2,
      2,
      0,
      0,
      0,
    ],
    Array [
      0,
      2,
      0,
      0,
      2,
    ],
    Array [
      2,
      2,
      0,
      2,
      2,
    ],
    Array [
      0,
      2,
      2,
      0,
      0,
    ],
    Array [
      2,
      0,
      2,
      0,
      0,
    ],
    Array [
      2,
      2,
      2,
      0,
      2,
    ],
    Array [
      0,
      0,
      2,
      2,
      0,
    ],
    Array [
      0,
      2,
      0,
      2,
      0,
    ],
    Array [
      2,
      2,
      2,
      2,
      0,
    ],
    Array [
      0,
      0,
      0,
      2,
      2,
    ],
  ],
}
`;

exports[`364.2 should solve example 10,12,41,62,31 1`] = `
Object {
  "counter": 30,
  "steps": Array [
    Array [
      10,
      12,
      41,
      62,
      31,
    ],
    Array [
      2,
      29,
      21,
      31,
      21,
    ],
    Array [
      27,
      8,
      10,
      10,
      19,
    ],
    Array [
      19,
      2,
      0,
      9,
      8,
    ],
    Array [
      17,
      2,
      9,
      1,
      11,
    ],
    Array [
      15,
      7,
      8,
      10,
      6,
    ],
    Array [
      8,
      1,
      2,
      4,
      9,
    ],
    Array [
      7,
      1,
      2,
      5,
      1,
    ],
    Array [
      6,
      1,
      3,
      4,
      6,
    ],
    Array [
      5,
      2,
      1,
      2,
      0,
    ],
    Array [
      3,
      1,
      1,
      2,
      5,
    ],
    Array [
      2,
      0,
      1,
      3,
      2,
    ],
    Array [
      2,
      1,
      2,
      1,
      0,
    ],
    Array [
      1,
      1,
      1,
      1,
      2,
    ],
    Array [
      0,
      0,
      0,
      1,
      1,
    ],
    Array [
      0,
      0,
      1,
      0,
      1,
    ],
    Array [
      0,
      1,
      1,
      1,
      1,
    ],
    Array [
      1,
      0,
      0,
      0,
      1,
    ],
    Array [
      1,
      0,
      0,
      1,
      0,
    ],
    Array [
      1,
      0,
      1,
      1,
      1,
    ],
    Array [
      1,
      1,
      0,
      0,
      0,
    ],
    Array [
      0,
      1,
      0,
      0,
      1,
    ],
    Array [
      1,
      1,
      0,
      1,
      1,
    ],
    Array [
      0,
      1,
      1,
      0,
      0,
    ],
    Array [
      1,
      0,
      1,
      0,
      0,
    ],
    Array [
      1,
      1,
      1,
      0,
      1,
    ],
    Array [
      0,
      0,
      1,
      1,
      0,
    ],
    Array [
      0,
      1,
      0,
      1,
      0,
    ],
    Array [
      1,
      1,
      1,
      1,
      0,
    ],
    Array [
      0,
      0,
      0,
      1,
      1,
    ],
  ],
}
`;

exports[`364.2 should solve example 10,12,41,62,31,50 1`] = `
Object {
  "counter": 22,
  "steps": Array [
    Array [
      10,
      12,
      41,
      62,
      31,
      50,
    ],
    Array [
      2,
      29,
      21,
      31,
      19,
      40,
    ],
    Array [
      27,
      8,
      10,
      12,
      21,
      38,
    ],
    Array [
      19,
      2,
      2,
      9,
      17,
      11,
    ],
    Array [
      17,
      0,
      7,
      8,
      6,
      8,
    ],
    Array [
      17,
      7,
      1,
      2,
      2,
      9,
    ],
    Array [
      10,
      6,
      1,
      0,
      7,
      8,
    ],
    Array [
      4,
      5,
      1,
      7,
      1,
      2,
    ],
    Array [
      1,
      4,
      6,
      6,
      1,
      2,
    ],
    Array [
      3,
      2,
      0,
      5,
      1,
      1,
    ],
    Array [
      1,
      2,
      5,
      4,
      0,
      2,
    ],
    Array [
      1,
      3,
      1,
      4,
      2,
      1,
    ],
    Array [
      2,
      2,
      3,
      2,
      1,
      0,
    ],
    Array [
      0,
      1,
      1,
      1,
      1,
      2,
    ],
    Array [
      1,
      0,
      0,
      0,
      1,
      2,
    ],
    Array [
      1,
      0,
      0,
      1,
      1,
      1,
    ],
    Array [
      1,
      0,
      1,
      0,
      0,
      0,
    ],
    Array [
      1,
      1,
      1,
      0,
      0,
      1,
    ],
    Array [
      0,
      0,
      1,
      0,
      1,
      0,
    ],
    Array [
      0,
      1,
      1,
      1,
      1,
      0,
    ],
    Array [
      1,
      0,
      0,
      0,
      1,
      0,
    ],
    Array [
      1,
      0,
      0,
      1,
      1,
      1,
    ],
  ],
}
`;
